src/ShipBattleGame.java

```diff
@@ -0,0 +1,183 @@
+            default -> new int[]{-1, 0}; // to the west
+        };
+    }
+
+    // This method checks where it can be created and returns the coordinates that will be a ship's particles.
+    public static int[] check(int[][] map, int n) {
+        while (true) {
+            int[] coordinates = randomPoint();
+            int x = coordinates[0]; // starting x
+            int y = coordinates[1]; // starting y
+            int[] d = destination();
+            int i = d[0]; // x's increment
+            int j = d[1]; // y's increment
+            int[] willPaint = new int[n]; // ship's particles
+            int partOfShip = 0;
+            while (partOfShip != n) { // n is equal to the length of the ship
+                // checking if x and y is right place for to be ship's particle
+                if ((getValue(x, y, map) == 0) && (getValue(x + 1, y, map) != 1) && (getValue(x - 1, y, map) != 1) && (getValue(x, y + 1, map) != 1) && (getValue(x, y - 1, map) != 1)) {
+                    willPaint[partOfShip] = 10 * x + y; // i don't wanna add x and y as in array so i made them 2 digit number
+                    x += i;
+                    y += j;
+                    partOfShip++;
+                    if (partOfShip == n) { // if all particles are determined then return all particles
+                        return willPaint;
+                    }
+                } else { // if coordinates isn't right place exit ,and it will create new x and y
+                    break;
+                }
+            }
+        }
+    }
+
+    // getting value of coordinate if it's out of boundary it will give -2 else, it will give value of coordinate
+    public static int getValue(int i, int j, int[][] matrix) {
+        if (i < 0 || j < 0 || i > N - 1 || j > N - 1) {
+            return -2;
+        }
+        return matrix[i][j];
+    }
+
+    // Method for printing map, it is only for checking
+    public static void printArray(int[][] map) {
+        for (int[] row : map) {
+            for (int i : row) {
+                System.out.print(i + "  ");
+            }
+            System.out.println();
+        }
+    }
+
+    // Method for getting all coordinates of ships, this also is only for checking
+    public static void printArray(String[][] map) {
+        for (String[] row : map) {
+            for (String s : row) {
+                System.out.print(s + "  ");
+            }
+            System.out.println();
+        }
+    }
+
+    // This method will create random x and y values
+    public static int[] randomPoint() {
+        int x = RAND.nextInt(10);
+        int y = RAND.nextInt(10);
+        return new int[]{x, y};
+    }
+
+    // This method returns if the ship fully destroyed
+    public static boolean checkIfDestroyed(int[] pointCoordinate, String[][] allShips) {
+        for (int j = 0; j < allShips[pointCoordinate[0]].length; j++) {
+            if (!Objects.equals(allShips[pointCoordinate[0]][j], "DD")) {
```

Comment:
it checks only ship which is placed as row, Ship also can be placed as horizontal